### PR TITLE
Custom scanner implementation for Dialog Tag Language

### PR DESCRIPTION
**Pull request by dennis-behm**
DBB has an extensibility mechanism to contribute your own dependency scanner for languages that either is not covered by the DBB product itself or that you might use some custom syntax, that is custom to your environment. 

This contribution from @campagboy is outlining how to build a customer scanner using the [AbstractDependencyScanner](https://www.ibm.com/docs/api/v1/content/SS6T76_2.0.0/javadoc/com/ibm/dbb/dependency/AbstractDependencyScanner.html) API of DBB for the **Dialog Tag Language** in ISPF.

The DTL scanner comes with the java project, that you need to build with then add to the JAVA classpath. 

Also combining the default DependencyScanner is possible with your own parsing logic is possible by writing a new scanner that invokes the default [DependencyScanner](https://www.ibm.com/docs/api/v1/content/SS6T76_2.0.0/javadoc/com/ibm/dbb/dependency/DependencyScanner.html)  then adds your own dependencies.

### CODE REVIEW

File: Scanners/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Scanners
+
+This category contains examples how to build your custom dependency scanner based on DBBs' `AbstractDependencyScanner` API.
+Samples are supposed to follow the same structure. The custom dependency scanner may either be a scanner implementation by itself or extend an existing scanner.
+
+## Building a scanner with IDZ
+
+* Clone the IBM/DBB repository into IDZ/Eclipse.
+* Import the scanner implementation as a JAVA project
+* Download the `<DBB-HOME>/lib/dbb.core_2.0.x.x.jar` package from your IBM DBB toolkit installation to your development environment
```
Comment (M-DLB) [inline Scanners/README.md:10]: Maybe we should explicitly say "from your IBM DBB toolkit installation on z/OS Unix System Services to..."

File: Scanners/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+* Import the scanner implementation as a JAVA project
+* Download the `<DBB-HOME>/lib/dbb.core_2.0.x.x.jar` package from your IBM DBB toolkit installation to your development environment 
+* Fix the build setup and of the scanner project.
+* Implement changes in `runScan()` method of the scanner implementation that is extending `AbstractDependencyScanner`.
+* Create a `Run Configuration` for a Java Application for the scanner implementation (Note: The Run Configuration will not have a main method as we will not actually be using it to run the program. You might see some warnings.)
+* Execute the `Export Java > Runnable JAR file` action in IDZ/Eclipse, which will create the JAR file containing your custom scanner:
+  * Select the previously created runner configuration
+  * Select the export destination
+  * In the Library handling section, select to `Copy required libraries into a sub-folder next to the generated JAR` option
+* Upload the generated JAR to z/OS Unix system services
```
Comment (M-DLB) [inline Scanners/README.md:18]: "Copy the generated JAR to z/OS Unix system services as binary"

File: .project [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+<projectDescription>
+	<name>dbb</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+	</buildSpec>
+	<natures>
+	</natures>
+</projectDescription>
```
Comment (M-DLB) [inline .project:11]: Do we need to have a .project for the whole dbb repository?

File: Scanners/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+This category contains examples how to build your custom dependency scanner based on DBBs' `AbstractDependencyScanner` API.
+Samples are supposed to follow the same structure. The custom dependency scanner may either be a scanner implementation by itself or extend an existing scanner.
+
+## Building a scanner with IDZ
+
+* Clone the IBM/DBB repository into IDZ/Eclipse.
+* Import the scanner implementation as a JAVA project
+* Download the `<DBB-HOME>/lib/dbb.core_2.0.x.x.jar` package from your IBM DBB toolkit installation to your development environment 
+* Fix the build setup and of the scanner project.
```
[the remark this answers is not in the document]
Comment (dennis-behm) [reply Scanners/README.md]: Fix in 2adfcc551f9c66f28ef3f658a8162c3841f01a9f

File: Scanners/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+* Copy the generated JAR to z/OS Unix system services as binary
+* Run the provided test script which each scanner implementation should provide. Details are included in the scanner project.
+ 
+## Scanner implementations
+
+The below list contains the available scanner implementations
+
+Sample | Description
+--- | ---
+[DTL Scanner](DTL%20Scanner) | A custom scanner implementation for z/OS ISPF Dialog Tag Language to detect it's dependencies. zAppBuild contains the language script for DTL. The scanner implementation leverages regex patterns to identify the referenced include files.
```
Comment (M-DLB) [inline Scanners/README.md:27]: Small typo here: `it's dependencies` --> `its dependencies`
Sorry I missed that.

**Review by M-DLB — COMMENTED** [review]
Sorry I missed one typo in my previous review.

**Pull request merged**